src/Microsoft.AspNetCore.Mvc.Razor.Extensions/RazorExtensionsDiagnosticFactory.cs

```diff
@@ -98,5 +98,17 @@ public static RazorDiagnostic CreateViewComponent_SyncMethod_CannotReturnTask(st
 
             return diagnostic;
         }
+
+        public static readonly RazorDiagnosticDescriptor PageDirective_CannotBeImported =
+            new RazorDiagnosticDescriptor(
+                $"{DiagnosticPrefix}3905",
+                () => Resources.PageDirectiveCannotBeImported,
```

Comment:
That's the signature of the diagnostic descriptor. Essentially it enables errors to have scoped values intertwined (taken from Roslyn).